### PR TITLE
Reorganize cookbook

### PR DESCRIPTION
**Pull request by MMBazel**
# TODOs
The tl;dr: 

* The priority is:
> * First make sure notebooks and apps can run locally, with minimal setup (i.e. just using a virtual environment)
> * Then enable running in a cloud-based environment (like colab, HF spaces, or lightning AI studios)
> * Then enable containers

* Each project will have its own dependencies and for the purpose of having a clean contribution and execution experience, we will not have shared dependences across projects.
* Running a notebook example or an app should be consistent i.e. either all projects come with Dockerfiles, etc.
* Additional details in [internal-gdoc](https://docs.google.com/document/d/1l9oeMOrFClZfNsgo_FO1KoXj4htxuURC5UpfPJAgAT0/edit?usp=sharing)

## What I've done in this PR: 

**Pull-in projects**
- [x] Pull-in projects from Small Data SF Workshop repo
- [x] Pull-in projects from Forge repo
- [x] Pull-in projects from ai-starter-kits

**Created structure via subdirectories and readme's**
- [x] Similar to the [Pinecone examples repo](https://github.com/pinecone-io/examples), I created separate sections for FW-team projects & learning notebooks and apps. 
- [x] I also created a section for partner collateral in `integrations` as well as `showcase` for any user generated projects and examples.
- [x] Added `Contribution`  instructions. 
- [x] A preview of how it looks is in this [`examples` repo](https://github.com/fw-ai/examples). 
- [x]  And for now I've moved the previous contents of the cookbook into a subdirectory called `old_cookbook`. There are different types of examples projects in the old cookbook with different shared dependencies. For the next PR, I will go through each of the projects, test the notebooks locally, and make any adjustments so they can run locally in a virtual environment (not just Colab) and then move them to the right directory and list them in the README.md. 

## What will be in the next PR

Where I want to get to is this really clean, cohesive experience: [https://docs.pinecone.io/home](https://docs.pinecone.io/home) with: `examples`, `troubleshooting`, `releases` at the top.

**Generate the corresponding mdx files for the docs repo**
* Note: That work will not be in this repo but the files shouldn't be moved around, as those mdx files will link out to projects here. 

**Re-organize & clean up old cookbook projects**
- [ ]  See above comment 

**Archive the repos**
* Note: This isn't work for this repo. 
- [ ] Data SF Workshop repo
- [ ] Forge repo
- [ ] ai-starter-kits
- [ ] examples repo

**Re-share cookbook**
- [ ] Discord - Cookbook should be pinned in the `Getting Started` channel on Discord.
- [ ] Twitter - We don't promote the cookbook directly, instead indirectly by promoting new projects we've added.
- [ ] LinkedIn - We don't promote the cookbook directly, instead indirectly by promoting new projects we've added.

### CODE REVIEW

File: learn/function-calling/project_using_firefunction_v1/.gitignore [hunk 1]
```diff
@@ -0,0 +1,6 @@
+/fc
```
Comment (pgarbacki) [inline learn/function-calling/project_using_firefunction_v1/.gitignore:1]: Where did you copy `project_using_firefunction_v1` from?

File: old_cookbook/.gitignore [hunk 2]
```diff
@@ -0,0 +1,160 @@
+# Byte-compiled / optimized / DLL files
```
Comment (pgarbacki) [inline old_cookbook/.gitignore:1]: Should we just remove `old_cookbook`?

Comment (MMBazel) [reply old_cookbook/.gitignore:1]: Not yet, there are various notebooks and apps in the old notebook that either didn't have requirements.txt files, etc. I don't want to remove them but at the same time I want to clear a space so partners can start contribution notebooks and the .mdx files

File: learn/function-calling/project_using_firefunction_v1/.gitignore (hunk 1 above)
Comment (MMBazel) [reply learn/function-calling/project_using_firefunction_v1/.gitignore:1]: I copied it from here: https://github.com/fw-ai/ai-starter-kits/tree/main/function-calling/firefunction

It's supposed to be the code for this blog post: https://fireworks.ai/blog/function-call-vercel-fastapi-serp
But some of the the links are broken in the blog post, so it took a bit of digging to confirm. 

The project doesn't look like it was fully-setup to be run locally - so I'm not 100% what would be the best approach. 

option 1
- Add a ReadME (that has the salient info from the blog posts) 
- Adapt the instructions for running the project in a virtual env locally
Then leave it add that

option 2
???

Comment (pgarbacki) [reply learn/function-calling/project_using_firefunction_v1/.gitignore:1]: option 2 sgtm

**Review by pgarbacki — APPROVED** [review]
